doc/tunnel/SAI-Proposal-EVPN-Multihoming.md

```diff
@@ -0,0 +1,616 @@
+
+
+# 1.0  Introduction
+
+This document describes the SAI changes required for supporting EVPN Multihoming based on [RFC 7432](https://datatracker.ietf.org/doc/html/rfc7432) and [RFC 8365](https://datatracker.ietf.org/doc/html/rfc8365).
+
+RFC 7432 describes EVPN-MH support for MPLS networks and RFC 8365 describes the support for overlay networks like VxLAN.
+The scope of this document is EVPN MH support for VxLAN networks.
+ 
+## 1.1 Acronyms
+
+| Acronym   | Description      |
+|:----------|:----------|
+| BUM      | Broadcast, Unknown unicast, Multicast     |
+| DF       | Designated Forwarder                      |
+| ES       | Ethernet Segment                          |
+| EVPN     | Ethernet Virtual Private Network          |
+| MH       | Multihoming                               |
+| VNI      | Virtual Network Identifier                |
+| VTEP     | VXLAN Tunnel End point                    |
+
+
+# 2.0 EVPN-MH SAI Requirements
+
+1. It shall be possible to create/delete a group of remote VTEPs.
+2. It shall be possible to add/remove a remote VTEP to the remote VTEP group.
+3. It shall be possible to create MAC addresses with the destination as the remote VTEP group. 
+4. It shall be possible to set the MAC address destination as a remote VTEP group.
+5. It shall be possible to implement the DF functionality at a per port basis. 
+6. It shall be possible to implement the split horizon functionality as described in RFC 7432.
+8. It shall be possible to implement single active redundancy mode as described in RFC 7432.
+7. It shall be possible to implement a fast failover in the event of an ES going down.
+
+# 3.0 EVPN-MH SAI Components
+
+## 3.1 New SAI Objects 
+No new SAI objects are introduced as part of this design.
+
+## 3.2 Changes to Existing SAI Objects
+
+### 3.2.1 Remote VTEP group support
+
+A new nexthop group type to allow for MACs to point to the remote VTEP group is added.
+
+```
+typedef enum _sai_next_hop_group_type_t
+...
+    /** Next hop group is for bridge port */
+    SAI_NEXT_HOP_GROUP_TYPE_BRIDGE_PORT,
+
+} sai_next_hop_group_type_t;
+```
+
+A new nexthop type which will be part of groups of the above group type.
+
+```
+typedef enum _sai_next_hop_type_t
+...
+    /** Next hop group is for bridge port */
+    SAI_NEXT_HOP_TYPE_BRIDGE_PORT,
+} sai_next_hop_type_t;
+```
+
+The above nexthop type will need ATTR_TUNNEL_ID as well as ATTR_IP as mandatory parameters on creation.
+
+```
+typedef enum _sai_next_hop_attr_t
+...
+    /**
+     * @brief Next hop entry IPv4 address
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_IP or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_MPLS or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_TUNNEL_ENCAP or SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_BRIDGE_PORT
+     */
+    SAI_NEXT_HOP_ATTR_IP,

```

Comment:
Do we need attr SAI_NEXT_HOP_ATTR_IP for SAI_NEXT_HOP_TYPE_BRIDGE_PORT? Is this for SAI_TUNNEL_PEER_MODE_P2MP?

if yes, please mention this in comment.

Reply:
Yes for P2MP.